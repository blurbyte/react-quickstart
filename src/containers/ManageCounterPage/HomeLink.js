import NavLink from '../../components/NavLink';
import styled from 'styled-components';

const HomeLink = styled(NavLink)`
  margin-top: 4rem;
`;

export default HomeLink;
