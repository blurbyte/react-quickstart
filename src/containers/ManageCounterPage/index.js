import React from 'react';
import Helmet from 'react-helmet';

//components
import ContentWrapper from '../../components/ContentWrapper';
import MainHeadline from '../../components/MainHeadline';
import MainParagraph from '../../components/MainParagraph';
import HomeLink from './HomeLink';

import ManageCounter from '../ManageCounter';

const ManageCounterPage = () => (
  <ContentWrapper>
    <Helmet title="Manage Counter" />
    <MainHeadline>Manage Counter</MainHeadline>
    <MainParagraph>It's simple Redux counter example. Current counter value is held in Redux Store. Counter shown in header and on this page below are connected to Redux Store &mdash; value got updated instantly in both places. It also persists even if you move between pages.</MainParagraph>
    <ManageCounter />
    <HomeLink to="/">Go back to main page</HomeLink>
  </ContentWrapper>
);

export default ManageCounterPage;
