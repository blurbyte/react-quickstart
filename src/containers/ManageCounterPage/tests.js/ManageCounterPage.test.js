import React from 'react';
import renderer from 'react-test-renderer';

import ManageCounterPage from '../index';

// mocked components
jest.mock('react-helmet', () => 'Helmet');
jest.mock('components/ContentWrapper', () => 'ContentWrapper');
jest.mock('components/MainHeadline', () => 'MainHeadline');
jest.mock('components/MainParagraph', () => 'MainParagraph');
jest.mock('../HomeLink', () => 'HomeLink');
jest.mock('containers/ManageCounter', () => 'ManageCounter');

describe('<ManageCounterPage />', () => {
  it('should render correctly', () => {
    // component init
    const component = renderer.create(<ManageCounterPage />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
