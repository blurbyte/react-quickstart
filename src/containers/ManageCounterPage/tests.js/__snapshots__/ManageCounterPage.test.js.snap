// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManageCounterPage /> should render correctly 1`] = `
<ContentWrapper>
  <Helmet>
    <title>
      Manage Counter
    </title>
  </Helmet>
  <MainHeadline>
    Manage Counter
  </MainHeadline>
  <MainParagraph>
    It's simple Redux counter example. Current counter value is held in Redux Store. Counter shown in header and on this page below are connected to Redux Store â€” value got updated instantly in both places. It also persists even if you move between pages.
  </MainParagraph>
  <ManageCounter />
  <HomeLink
    to="/"
  >
    Go back to main page
  </HomeLink>
</ContentWrapper>
`;
